Design new .NET Core Setup/Install area; Rewrite prereqs articles; Add package managers (#15569)
dotnet/docs: docs/core/setup/how-to-detect-installed-versions.md

---

> title: Check which .NET Core versions on Windows, Linux, and macOS

Reviewer: ```suggestion
title: Check which .NET Core versions on Windows, Linux, and macOS are installed - .NET Core
```

Author: Should we just add this folder to the `titleSuffix` entry in docfx.json?

description: Learn how to list which versions of .NET Core are installed on your computer. This includes the .NET Core runtime and SDK.
author: thraka
ms.author: adegeo
ms.date: 11/01/2019
zone_pivot_groups: operating-systems-set-one
---

# How to check if .NET Core is already installed

> 

Reviewer: Should you make clear that this article requires that you have the SDK installed already? And how would you check which versions of the runtime you have installed without having the SDK installed?

Author: you don't need the SDK to use these commands.

Reviewer: I didn't know you could do this just with the runtime!

So should you mention that if the commands fail to work it means you don't have any version installed?

Author: Good idea!

Reviewer: It also requires 2.0+ since those options AFAIK were only introduced in 2.0.

Author: 2.0 is the lowest supported .NET Core version.

> This article teaches you how to check which versions of the .NET Core runtime and SDK are installed on your computer. If you have a development IDE, such as Visual Studio or Visual Studio for Mac, .NET Core may have been installed.

Reviewer: define IDE

> Installing an SDK will install the corresponding runtime.

Reviewer: ```suggestion
Installing an SDK installs the corresponding runtime.
```

> ## Runtimes

Reviewer: ```suggestion
## SDK
```

You can see which versions of the .NET Core SDK are currently installed with a terminal. Open a terminal and run the `dotnet --list-sdks` command.

::: zone pivot="os-windows"

```console
dotnet --list-sdks

2.1.500 [C:\program files\dotnet\sdk]
2.1.502 [C:\program files\dotnet\sdk]
2.1.504 [C:\program files\dotnet\sdk]
2.1.600 [C:\program files\dotnet\sdk]
2.1.602 [C:\program files\dotnet\sdk]
2.2.101 [C:\program files\dotnet\sdk]
3.0.100 [C:\program files\dotnet\sdk]
```

::: zone-end

::: zone pivot="os-linux"

```bash
dotnet --list-sdks

2.1.500 [/home/user/dotnet/sdk]
2.1.502 [/home/user/dotnet/sdk]
2.1.504 [/home/user/dotnet/sdk]
2.1.600 [/home/user/dotnet/sdk]
2.1.602 [/home/user/dotnet/sdk]
2.2.101 [/home/user/dotnet/sdk]
3.0.100 [/home/user/dotnet/sdk]
```

::: zone-end

::: zone pivot="os-macos"

```bash
dotnet --list-sdks

2.1.500 [/usr/local/share/dotnet/sdk]
2.1.502 [/usr/local/share/dotnet/sdk]
2.1.504 [/usr/local/share/dotnet/sdk]
2.1.600 [/usr/local/share/dotnet/sdk]
2.1.602 [/usr/local/share/dotnet/sdk]
2.2.101 [/usr/local/share/dotnet/sdk]
3.0.100 [/usr/local/share/dotnet/sdk]
```

::: zone-end

> ## SDKs

Reviewer: ```suggestion
## Runtime
```

> You can see which versions of the .NET Core SDK are currently installed with the `dotnet --list-sdks` command.

Reviewer: ```suggestion
You can see which versions of the .NET Core runtime are currently installed with the `dotnet --list-runtimes` command.
```

> You can see which versions of the .NET Core SDK are currently installed by using the `dotnet --list-sdks` command.

Reviewer: ```suggestion
```

::: zone pivot="os-windows"

```console

> dotnet --list-sdks

Reviewer: ```suggestion
dotnet --list-runtimes
```

Microsoft.AspNetCore.App 2.1.6 [c:\program files\dotnet\shared\Microsoft.AspNetCore.App]
Microsoft.AspNetCore.App 2.1.7 [c:\program files\dotnet\shared\Microsoft.AspNetCore.App]
Microsoft.AspNetCore.App 2.1.13 [c:\program files\dotnet\shared\Microsoft.AspNetCore.App]
Microsoft.AspNetCore.App 2.2.0 [c:\program files\dotnet\shared\Microsoft.AspNetCore.App]
Microsoft.AspNetCore.App 2.2.7 [c:\program files\dotnet\shared\Microsoft.AspNetCore.App]
Microsoft.AspNetCore.App 3.0.0 [c:\program files\dotnet\shared\Microsoft.AspNetCore.App]
Microsoft.NETCore.App 2.1.7 [c:\program files\dotnet\shared\Microsoft.NETCore.App]
Microsoft.NETCore.App 2.1.13 [c:\program files\dotnet\shared\Microsoft.NETCore.App]
Microsoft.NETCore.App 2.2.0 [c:\program files\dotnet\shared\Microsoft.NETCore.App]
Microsoft.NETCore.App 2.2.3 [c:\program files\dotnet\shared\Microsoft.NETCore.App]
Microsoft.NETCore.App 2.2.7 [c:\program files\dotnet\shared\Microsoft.NETCore.App]
Microsoft.NETCore.App 3.0.0 [c:\program files\dotnet\shared\Microsoft.NETCore.App]
Microsoft.WindowsDesktop.App 3.0.0 [c:\program files\dotnet\shared\Microsoft.WindowsDesktop.App]
```

::: zone-end

::: zone pivot="os-linux"

```bash

> dotnet --list-sdks

Reviewer: ```suggestion
dotnet --list-runtimes
```

Microsoft.AspNetCore.All 2.2.7 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.All]
Microsoft.AspNetCore.App 2.1.6 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
Microsoft.AspNetCore.App 2.1.7 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
Microsoft.AspNetCore.App 2.1.13 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
Microsoft.AspNetCore.App 2.2.0 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
Microsoft.AspNetCore.App 2.2.7 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
Microsoft.AspNetCore.App 3.0.0 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
Microsoft.NETCore.App 2.1.7 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
Microsoft.NETCore.App 2.1.13 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
Microsoft.NETCore.App 2.2.0 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
Microsoft.NETCore.App 2.2.3 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
Microsoft.NETCore.App 2.2.7 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
Microsoft.NETCore.App 3.0.0 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
```

::: zone-end

## More information

> You can see both the SDK versions and runtime versions with the command `dotnet --info`. You'll also get a lot of other environmental related information, such as the operating system version and runtime-identifier (RID).

Reviewer: ```suggestion
You can see both the SDK versions and runtime versions with the command `dotnet --info`. You'll also get a lot of other environmental related information, such as the operating system version and runtime identifier (RID).
```

## Next steps

> 

Reviewer: ?

---
title: Check installed .NET Core versions on Windows, Linux, and macOS - .NET Core
description: Learn how to list which versions of .NET Core are installed on your computer. This includes the .NET Core runtime and SDK.
author: thraka
ms.author: adegeo
ms.date: 11/01/2019
ms.custom: "updateeachrelease"
zone_pivot_groups: operating-systems-set-one
---

> # How to verify .NET Core is already installed

Reviewer: I'm going back on this. Check seems to have more search results.

> # How to check if .NET Core is already installed

Reviewer: Would verify be better?

```suggestion
# How to verify if .NET Core is already installed
```

This article teaches you how to check which versions of the .NET Core runtime and SDK are installed on your computer. If you have an integrated development environment, such as Visual Studio or Visual Studio for Mac, .NET Core may have been installed.

Installing an SDK installs the corresponding runtime.

> ## SDK

Reviewer: should you add a verb here?

```suggestion
## Check SDK versions
```

You can see which versions of the .NET Core SDK are currently installed with a terminal. Open a terminal and run the `dotnet --list-sdks` command.

::: zone pivot="os-windows"

```console
dotnet --list-sdks

2.1.500 [C:\program files\dotnet\sdk]
2.1.502 [C:\program files\dotnet\sdk]
2.1.504 [C:\program files\dotnet\sdk]
2.1.600 [C:\program files\dotnet\sdk]
2.1.602 [C:\program files\dotnet\sdk]
2.2.101 [C:\program files\dotnet\sdk]
3.0.100 [C:\program files\dotnet\sdk]
```

::: zone-end

::: zone pivot="os-linux"

```bash
dotnet --list-sdks

2.1.500 [/home/user/dotnet/sdk]
2.1.502 [/home/user/dotnet/sdk]
2.1.504 [/home/user/dotnet/sdk]
2.1.600 [/home/user/dotnet/sdk]
2.1.602 [/home/user/dotnet/sdk]
2.2.101 [/home/user/dotnet/sdk]
3.0.100 [/home/user/dotnet/sdk]
```

::: zone-end

::: zone pivot="os-macos"

```bash
dotnet --list-sdks

2.1.500 [/usr/local/share/dotnet/sdk]
2.1.502 [/usr/local/share/dotnet/sdk]
2.1.504 [/usr/local/share/dotnet/sdk]
2.1.600 [/usr/local/share/dotnet/sdk]
2.1.602 [/usr/local/share/dotnet/sdk]
2.2.101 [/usr/local/share/dotnet/sdk]
3.0.100 [/usr/local/share/dotnet/sdk]
```

::: zone-end

> ## Runtime

Reviewer: ```suggestion
## Check Runtime versions
```
